packages/base-types/src/utils/ai.ts

```diff
@@ -31,6 +31,11 @@ export interface AIModelParams {
   system?: string;
 }
 
+export interface AICompletionParams {
+  // make it optional for backward compatibility
+  messages?: Message[];
```

Comment:
it's better to store only references/ids in the step/node data. We have access to all prompt messages in the runtime and we can get them by the surveyor context. Might be better to actually only store the `promptID`

Reply:
Actually since this step data is before compile, we can use the existent `responsePrompt`

Reply:
hmmm this is for the node data (AI_RESPONSE) after compilation.

This node is created during compilation, transforming the `promptID` to an actual runtime compatible AI_RESPONSE node. This nodes currently supports system/prompt, so passing a `promptID` wont cut it, we would need to move a bunch of logic to the runtime side.

This approach makes the compilation act kind of the "adapter" layer, the runtime keeps functioning as it is, with very minor changes to support it being more extensible.

Reply:
got it, makes sense. Im asking because you added it to `step data` [here](https://github.com/voiceflow/libs/pull/551/files#diff-3ccc6584087f15d4d7ff0b9006d3c034ca0f7b5cf734f7694db629ec81cdd9a0R13) that is supposed to be before compilation 🤔 Shouldn't it be added to the node data?

Reply:
oh this is being share by both the node and the step 😞 

ill see if I can decouple it

Reply:
oh this is being share by both the node and the step 😞 

ill see if I can decouple it